### PR TITLE
ENG-11043: EE exceptions doing updates on MV

### PR DESCRIPTION
**Pull request by wolffcm**
Fix for ENG-11043, which was an EE issue with loop counters on min/max aggs.

Also made it so assertTablesAreEqual does what it's supposed to.  I added an optional epsilon argument for the places that use it.

Fix assertTablesAreEqual turned up another bug that went uncaught in TestMaterializedViewSuite, for which I have filed ENG-11048.

### CODE REVIEW

File: src/ee/storage/MaterializedViewHandler.cpp [hunk 1]
```diff
@@ -310,13 +312,18 @@ namespace voltdb {
...
                         assert(false); // Should have been caught when the matview was loaded.
                         // no break
                 }
             }
+
+            if (aggType == EXPRESSION_TYPE_AGGREGATE_MIN
+                || aggType == EXPRESSION_TYPE_AGGREGATE_MAX) {
+                minMaxColumnIndex++;
+            }
+
```
Comment (BillWhite) [inline src/ee/storage/MaterializedViewHandler.cpp]: So the only semantic difference here is that minMaxColumnIndex is incremented even if deltaValue is NULL.  Is that right?

**Comment by BillWhite** [comment]
I think  I understand why the old code would have failed, and why this fixes it.

So, it seems right to me.

File: src/ee/storage/MaterializedViewHandler.cpp (hunk 1 above)
Comment (wolffcm) [inline src/ee/storage/MaterializedViewHandler.cpp]: Yes, that's it exactly.

**Comment by wolffcm** [comment]
Closing so I can merge to release-6.6.x instead.

**Pull request closed without merging**